src/cascadia/TerminalSettingsModel/Command.cpp

```diff
@@ -69,6 +75,151 @@ namespace winrt::Microsoft::Terminal::Settings::Model::implementation
     {
         return _subcommands ? _subcommands.Size() > 0 : false;
     }
+
+    bool Command::HasName() const noexcept
+    {
+        return _name.has_value();
+    }
+
+    hstring Command::Name() const noexcept
+    {
+        if (_name.has_value())
+        {
+            // name was explicitly set, return that value.
+            return hstring{ _name.value() };
+        }
+        else if (_action)
+        {
+            // generate a name from our action
+            return get_self<ActionAndArgs>(_action)->GenerateName();
+        }
+        else
+        {
+            // we have no name
+            return {};
+        }
+    }
+
+    void Command::Name(const hstring& value)
+    {
+        if (!_name.has_value() || _name.value() != value)
+        {
+            _name = value;
+            _PropertyChangedHandlers(*this, Windows::UI::Xaml::Data::PropertyChangedEventArgs{ L"Name" });
+        }
+    }
+
+    std::vector<Control::KeyChord> Command::KeyMappings() const noexcept
+    {
+        return _keyMappings;
+    }
+
+    // Function Description:
+    // - Add the key chord to the command's list of key mappings.
+    // - If the key chord was already registered, move it to the back
+    //   of the line, and dispatch a notification that Command::Keys changed.
+    // Arguments:
+    // - keys: the new key chord that we are registering this command to
+    // Return Value:
+    // - <none>
+    void Command::RegisterKey(const Control::KeyChord& keys)
+    {
+        // Check if we registered this key chord before
+        for (auto pos = _keyMappings.begin(); pos < _keyMappings.end(); ++pos)
+        {
+            if (keys.Modifiers() == pos->Modifiers() && keys.Vkey() == pos->Vkey())
+            {
+                // KeyChord was already registered.
+                if (*pos == _keyMappings.back())
+                {
+                    // It's already the latest key registered.
+                    return;
+                }
+                else
+                {
+                    // Move the new KeyChord to the back of the line.
+                    _keyMappings.erase(pos);
+                    _keyMappings.push_back(*pos);
+                    // TODO CARLOS: tests fail because this is on the wrong thread
+                    //_PropertyChangedHandlers(*this, Windows::UI::Xaml::Data::PropertyChangedEventArgs{ L"Keys" });

```

Comment:
I'm just removing all of this property changed stuff for now. We're not using them since we can't modify `Command::Keys` (we just completely reconstruct a `CascadiaSettings`).

This _will_ be a problem when the Actions Page can edit `Command`s, but that's a problem for future Carlos haha.